user/test_dup/test_dup.c

```diff
@@ -0,0 +1,15 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+int main() {
+  int x = open("/tests/dup_test_file", O_RDONLY, 0);
+  int y = dup(0);
+  printf("x = %d, y = %d\n", x, y);
+  dup2(x, 0);
+  char word[100];
+  scanf("%[^\n]s", word);
+  printf("%s\n", word);
```

Comment:
I'd suggest to `assert(strcmp(word,"Hello, World!")==0)` so that we can verify the test without paying attention to the output.